wasp/src/main/java/com/orhanobut/wasp/parsers/Parser.java

```diff
@@ -7,7 +9,7 @@
  */
 public interface Parser {
 
-    <T> T fromJson(String content, Type type) throws Exception;
+    <T> T fromJson(String content, Type type) throws JsonSyntaxException;
```

Comment:
JsonSyntaxException creates a coupling here. We should either introduce our own Json exception or it should stay Exception.